operator-wg/whitepaper/051_kubebuilder.md

```diff
@@ -0,0 +1,17 @@
+## kubebuilder Operator Framework
+
+The kubebuilder framework provides developers the possibilities to extend the Kubernetes API by using Custom Resource Definitions, and to create operators that handle these custom resources.
+
+The main concept provided by the kubebuilder framework is a *Manager*. In the same way the native Kubernetes controllers are grouped into a single Kubernetes Controller Manager (`kube-controller-manager`), you will be able to create several operators and make them managed by a single manager.
+
+As Kubernetes API resources are attached to domains and arranged in Groups, Versions and Kinds, the Kubernetes custom resources you will define will be attached to your own domain, and arranged in your own groups, versions and kinds.
+
+The first step when using kubebuilder is to create a project, that  will create the source code for building a single Manager attached to your domain.
```

Comment:
Also a single manager can handle any number of API groups on any number of domains.

Reply:
Is it the normal behaviour of Kubebuilder, is it possible to handle several domains by using the CLI?

Reply:
Technically speaking, right now `""`, `"k8s.io"` and the domain you specify during init can be handled in the same project just by using the CLI. Additionally, kubebuilder is a scaffolder, meaning that it generates code but you have to edit and implement some parts, and changing the domain to a CRD is pretty easy and strightforward.

Outside of that, I think that what @coderanger means is that the manager is in no way bound to a single domain and this phrase is missleading in that sense.

Reply:
I agree that the manager is able to handle several domains, but this phrase, and more generally this section, are describing kubebuilder.
It can be confusing if the reader understands that kubebuilder gives the possibility to handle several domains, but no options are found at the kubebuilder level to do this.

Reply:
Maybe?
```suggestion
The first step when using kubebuilder is to create a project attached to your domain, that  will create the source code for building a single Manager.
```